pypeit/scripts/run_to_calibstep.py

```diff
@@ -0,0 +1,100 @@
+"""
+Script to run to a single calibration step for an input frame
+
+.. include common links, assuming primary doc root is up one directory
+.. include:: ../include/links.rst
+"""
+
+from pypeit.scripts import scriptbase
+
+class RunToCalibStep(scriptbase.ScriptBase):
+
+    # TODO: Combining classmethod and property works in python 3.9 and later
+    # only: https://docs.python.org/3.9/library/functions.html#classmethod
+    # Order matters.  In python 3.9, it would be:
+    #
+    # @classmethod
+    # @property
+    #
+    # Because we're not requiring python 3.9 yet, we have to leave this as a
+    # classmethod only:
+    @classmethod
+    def name(cls):
+        """
+        Return the name of the executable.
+        """
+        return 'pypeit_run_to_calibstep'
+
+    @classmethod
+    def get_parser(cls, width=None):
+        import argparse
+
+        parser = super().get_parser(description='Run PypeIt to a single calibration step for an input frame',
+                                    width=width, formatter=argparse.RawDescriptionHelpFormatter)
+        parser.add_argument('pypeit_file', type=str,
+                            help='PypeIt reduction file (must have .pypeit extension)')
+        parser.add_argument('frame', type=str, help='Raw science frame to reduce as listed in your PypeIt file, e.g. b28.fits.gz')
+        parser.add_argument('step', type=str, help='Calibration step to perform')
```

Comment:
It would be good to provide the list of possible steps here. Or, we could add another parameter, e.g. `--list_steps`, which will list all the default steps for that Calibrations class.

Reply:
Good suggestion.  I would define a class attribute:

```
    valid_steps = ['align', 'arc', 'bias', 'bpm', 'dark', 'flats', 'scattlight', 'slits', 'tiltimg', 'tilts', 'wv_calib']
```
or a class method:
```
    @classmethod
    def valid_steps(cls):
        steps = [method.replace('get_', '') for method in dir(cls) \
                    if callable(getattr(cls, method)) and method.startswith('get_')]
        steps.remove('instance')
        steps.remove('association')
        return steps
```
Then here, you would just do:
```
from pypeit.calibrations import Calibrations
parser.add_argument('step', type=str, help=f'Calibration step to perform.  Valid steps are {', '.join(Calibrations.valid_steps)}')
```
with or without the `()` for `valid_steps`, depending on the implementation you choose.